### PR TITLE
Allow multiple sponsored channels and randomize their sort.

### PR DESCRIPTION
**Pull request by madbook**
Removes limit on number of sponsored channels that can run at the same time, and also randomizes their sort.  Initially tried to do the randomization on the backend, but caching was an issue.

:eyeglasses: @Deimos

### CODE REVIEW

File: js/tv.js [hunk 1]
```diff
@@ -1875,6 +1875,18 @@ var RedditTV = Class.extend({
     }, // isiPad
 });
 
+// http://www.codinghorror.com/blog/2007/12/the-danger-of-naivete.html
+function shuffleArray(arr) {
+	var i, temp, j;
+	for (var i = 0, l = arr.length; i < l; i++) {
+		j = ~~(Math.random() * (i + 1));
```
Comment (ajacksified) [inline js/tv.js]: [parseint](https://github.com/reddit/styleguide/tree/master/javascript#type-casting--coercion)

Comment (madbook) [inline js/tv.js]: going to go with Math.floor, as I think its more explicit.  parseInt to me implies string -> int

Comment (ajacksified) [inline js/tv.js]: :+1:

**Comment by madbook** [comment]
:haircut:

File: js/tv.js [hunk 2]
```diff
@@ -1875,6 +1875,18 @@ var RedditTV = Class.extend({
     }, // isiPad
 });
 
+// http://www.codinghorror.com/blog/2007/12/the-danger-of-naivete.html
+function shuffleArray(arr) {
+	var i, temp, j;
```
Comment (ajacksified) [inline js/tv.js]: Since this is new code, vars should be declared on separate lines.

File: js/tv.js [hunk 3]
```diff
@@ -1875,6 +1875,18 @@ var RedditTV = Class.extend({
     }, // isiPad
 });
 
+// http://www.codinghorror.com/blog/2007/12/the-danger-of-naivete.html
+function shuffleArray(arr) {
+	var i, temp, j;
+	for (var i = 0, l = arr.length; i < l; i++) {
```
Comment (ajacksified) [inline js/tv.js]: You're re-`var`ing `i` here.

**Comment by ajacksified** [comment]
:nail_care:

**Comment by madbook** [comment]
:haircut:

**Comment by ajacksified** [comment]
:fish:

**Comment by madbook** [comment]
c8dd28726c8117837a2330173258fd52ab8ff438

**Pull request closed without merging**